client/client.go

```diff
@@ -89,6 +109,33 @@ func (c *vpnClient) Stop() error {
 	return err
 }
 
+// ConnectionStatus returns the connection status of the VPN client
+func (c *vpnClient) ConnectionStatus() bool {
+	clientMu.Lock()
+	defer clientMu.Unlock()
+	return c.started && c.connected
+}
+
+func (c *vpnClient) setConnectionStatus(connected bool) {
+	c.connected = connected
```

Comment:
since we're updating the `c.connnected` we probably need to lock 
```suggestion
        c.clientMu.Lock()
        defer clientMu.Unlock()
	c.connected = connected
```

Reply:
Thanks! I didn't actually use that function because of the strange goroutine so I completely missed that.